elements/lisk-transactions/test/8_transfer_transaction.spec.ts

```diff
@@ -335,4 +337,44 @@ describe('Transfer transaction class', () => {
 			expect(errors[0].message).toEqual('Invalid amount');
 		});
 	});
+
+	describe('#signAll', () => {
+		const { transaction, account, networkIdentifier } = validTransferInput;
+		let validTransferInstance: BaseTransaction;
+		beforeEach(async () => {
+			validTransferInstance = new TransferTransaction(transaction);
+		});
+
+		it('should have one signature for single key pair account', async () => {
+			validTransferInstance.signAll(
+				networkIdentifier,
+				account.passphrase,
+				undefined,
+				undefined,
```

Comment:
no need to pass the optional value?

Reply:
If it's a single signature transaction there's no need as pass-phrases and key sets are not used. For regular accounts the senderPublicKey is the one that signs.

Reply:
what I meant was, `undefined` was not required to pass you could just send the first two arguments and it would still work.